tests/unit_test/gmail_data/get_gmail_data_test.py

```diff
@@ -0,0 +1,219 @@
+    users_mock = gmail_service_mock.users
+    threads_mock = users_mock.return_value.threads
+    get_mock = threads_mock.return_value.get
+    return get_mock
+
+
+@pytest.fixture(name='gmail_history_list_mock')
+def _gmail_history_list_mock(gmail_service_mock: MagicMock) -> MagicMock:
+    users_mock = gmail_service_mock.users
+    history_mock = users_mock.return_value.history
+    list_mock = history_mock.return_value.list
+    return list_mock
+
+
+class TestGetLabelList:
+
+    def test_should_pass_user_id_to_list_method(
+            self,
+            gmail_service_mock,
+            gmail_label_list_mock):
+
+        get_label_list(gmail_service_mock, USER_ID1)
+        gmail_label_list_mock.assert_called_with(userId=USER_ID1)
+
+
+# pylint: disable=invalid-name
+class TestGetDataframeForLabelResponse:
+
+    def test_should_pass_user_id_and_imported_timestamp_in_method(self):
+        label_response = {"labels": [{"id": "id1", "name": "name1", "type": "type1"}]}
+        df = get_dataframe_for_label_response(label_response, USER_ID1, IMPORTED_TIMESTAMP)
+        assert df['user_id'].values == [USER_ID1]
+        assert df['imported_timestamp'].values == [IMPORTED_TIMESTAMP]
+
+    def test_should_return_elements_in_label_response(self):
+        label_response = {
+            "labels": [
+                {
+                    "id": "id1",
+                    "name": "name1",
+                    "type": "type1"
+                    },
+                {
+                    "id": "id2",
+                    "name": "name2",
+                    "type": "type2"
+                }
+            ]
+        }
+
+        df = get_dataframe_for_label_response(label_response, USER_ID1, IMPORTED_TIMESTAMP)
+        assert list(df['user_id']) == [USER_ID1, USER_ID1]
+        assert list(df['imported_timestamp']) == [IMPORTED_TIMESTAMP, IMPORTED_TIMESTAMP]
+        assert list(df['labelId']) == ['id1', 'id2']
+        assert list(df['labelName']) == ['name1', 'name2']
+        assert list(df['labelType']) == ['type1', 'type2']
+
+
+class TestIterLinkMessageThreadIds:
+
+    def test_should_pass_user_id_to_list_method(
+            self,
+            gmail_service_mock,
+            gmail_message_ids_list_mock):
+
+        list(iter_link_message_thread_ids(gmail_service_mock, USER_ID1))
+        gmail_message_ids_list_mock.assert_called_with(userId=USER_ID1)
+
+    def test_should_yield_first_page_messages_only_without_next_page_token(
+            self,
+            gmail_service_mock,
+            gmail_message_ids_list_mock):
+
+        response = {'messages': ['messages1', 'messages2']}
+
+        execute_mock = gmail_message_ids_list_mock.return_value.execute
+        execute_mock.return_value = response
+
+        actual_messages = list(iter_link_message_thread_ids(gmail_service_mock, USER_ID1))
+
+        assert actual_messages == ['messages1', 'messages2']
+
+    def test_should_next_page_messages_if_exists_next_page_token(
```

Comment:
naming is hard...
just because this is meant to read like English and when I read it in my non-native head it is not quite flowing I would maybe suggest to rename it just slightly.
e.g. "should" is usually followed by a verb.
perhaps something like `test_should_include_next_page_messages_if_next_page_token_exists`